src/cli/guess.rs

```diff
@@ -131,8 +135,12 @@ macro_rules! run_x {
                 };
                 if count > 0 {
                     println!("{}", v.name());
+                    guessed = true;
                 }
             }
+            if (!guessed) {
+                return Err(Error::InvalidType);
+            }
```

Comment:
This looks like a change that wasn't planned or discussed, and is unrelated to the change that's required for this issue. Could we move it into a separate pull request to discuss separately? Because the command returns a list that can be iterated on, returning an error in this case may make the command less composable. We might want to only return an error if `--exit-code` is passed.

The error would need renaming. Invalid type makes it sound like the user has specified a type. It may actually map to a valid type, but maybe a different input format needs selecting.

Reply:
Okay, let's do that!

Reply:
Created a task for that #413